### PR TITLE
[#5354] Update SSH connection method to use pka

### LINKED ISSUE #5354: Consider a public key login for the iRODS userzone management scripts
**Describe the feature you'd like and what it will do**

Currently, HS uses the following two key pairs in hydroshare/localsettings.py:

```
LINUX_ADMIN_USER_FOR_HS_USER_ZONE
LINUX_ADMIN_USER_PWD_FOR_HS_USER_ZONE
```

HS shells into the userzone ICAT with this user/pwd to run the` hydroshare/irods/create_user.sh` and `hydroshare/irods/delete_user.sh` scripts to create and delete iRODS users in the userzone.

Consider changing this to a public key login.

**Why is this feature important?**
A short description of the importance of this feature and what it will help you achieve.

In GCP, password authentication is disabled by default. This was changed in the GCP userzone to allow a password login so HS would continue to work. However, this is not really desirable.

**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...].

It is not ideal to sed `/etc/ssh/sshd_config` in the userzone to allow password logins.

**Additional context**

### PR DESCRIPTION
**Pull request by devincowan**
Resolves #5354

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Run `local-dev-first-start-only.sh`
2. See that userzone now functions for local development -- you can go to a user profile and create an irods (userzone) account, add a password, and it will create the account for you.
3. Remove the `LINUX_ADMIN_USER_PWD_FOR_HS_USER_ZONE` from your local_settings.py and instead use just the `PRIVATE_KEY_FILE_FOR_HS_USER_ZONE` and the `PUBLIC_KEY_FILE_FOR_HS_USER_ZONE`. See that the userzone auth works without password auth

### CODE REVIEW

**Comment by devincowan** [comment]
@cbcunc and @pkdash this is ready for your review.
The commit history is absolutely atrocious -- it took me a while to figure out how to get this to work correctly for PR builds. So we can squash the commits down once we have a final product.

File: hs_core/views/utils.py [hunk 1]
```diff
@@ -296,9 +296,16 @@ def run_ssh_command(host, uname, pwd, exec_cmd):
...
+    if not private_key_file and not pwd:
+        raise ValueError("Either Password or .pem is required for ssh connection")
     ssh = paramiko.SSHClient()
     ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-    ssh.connect(host, username=uname, password=pwd)
+    if private_key_file:
+        key = paramiko.Ed25519Key.from_private_key_file(private_key_file)
+        ssh.connect(host, username=uname, pkey=key)
+    else:
+        ssh.connect(host, username=uname, password=pwd)
```
Comment (devincowan) [inline hs_core/views/utils.py:307]: fallback to password if key not exist...

File: theme/views.py [hunk 2]
```diff
@@ -797,8 +811,8 @@ def create_irods_account(request):
                     return JsonResponse(
                         {
                             "error": "iRODS server failed to create this iRODS account {0}. "
-                            "If this issue persists, please notify help@cuahsi.org.".format(
-                                user.username
+                            "If this issue persists, please notify help@cuahsi.org. More details: {1}".format(
+                                user.username, out_str
```
Comment (devincowan) [inline theme/views.py:815]: show more info in error message when irods user create fails

**Comment by devincowan** [comment]
nevermind @cbcunc and @pkdash this PR has some issues I need to resolve before you review. Sorry for the false-start

**Comment by devincowan** [comment]
okay @pkdash and @cbcunc this is ready for review now

File: local-dev.yml [hunk 3]
```diff
@@ -105,25 +113,27 @@ services:
...
-    ports:
-      - "80:80"
-    volumes:
-      - "./log/nginx:/var/log/nginx"
-      - ".:/hydroshare"
-      - "temp_vol:/hs_tmp"
-      - "share_vol:/irods/cache"
-      - "idata_vault_vol:/irods/data"
-      - "iuser_vault_vol:/irods/user"
+  # TODO: nginx container does not currently work
```
Comment (devincowan) [inline local-dev.yml:116]: nginx has not been working on local dev for a long time. I removed it here because it takes time in the build process.

**Comment by cbcunc** [comment]
@devincowan You were going to test this on beta is what we last talked about in the dev meeting last week?

**Comment by cbcunc** [comment]
Also, formalize how this fits into deploy.sh and/or zdt.

**Comment by devincowan** [comment]
@cbcunc re `deploy.sh` I'm okay copying our key in to our hydroshare dir for deployment. But do we really want to have these keys floating around in backups etc?
I guess I don't see a problem with [just using a setting](https://github.com/hydroshare/hydroshare/pull/5403/files#diff-c65078e32ce609b4ab7a5fcb93c6b45c382a16412e1272e7211c3a971575f4c2R141):
`PRIVATE_KEY_FILE_FOR_HS_USER_ZONE = '/home/VM_USER_RUNNING_HS/.ssh/id_ed25519_hs'`
and placing the key file appropriately on all workers.

**Comment by devincowan** [comment]
@cbcunc I tested mounting the key using the docker-compose.yml on beta as you requested.
[Here is the commit](https://github.com/hydroshare/hydroshare/pull/5403/commits/561831ddd0584ca7776e8f3f7269ab27422ec120) where I "formalize how this fits into deploy.sh and/or zdt"

**Pull request merged**